### ISSUE TITLE
Trying to add multiple alphabetically adjacent imports from the same module results in assertion fail

### ISSUE THREAD

**Issue by nmggithub** [microsoft/TypeScript#42527]
# Bug Report

### 🔎 Search Terms

- add all missing imports
- debug failure
- false expression
- changes overlap

### 🕗 Version & Regression Information

This bug exists in v4.1.2 and v4.1.3 according to my testing.

- This is the behavior in every version I tried, and I reviewed the FAQ for entries about imports.

### ⏯ Playground Link

This is a bug with the TypeScript Server itself. It cannot be shown in the Playground.

### 💻 Code

This bug can be replicated in VSCode by making two files:

`exports.ts`
```ts
export interface Test1 {}
export interface Test2 {}
export interface Test3 {}
export interface Test4 {}
```
`importer.ts`
```ts
import { Test1, Test4 } from './exports';
interface Testing {
    test1: Test1;
    test2: Test2;
    test3: Test3;
    test4: Test4;
}
```

### 🙁 Actual behavior

Expectedly, the server correctly reports that it cannot find `Test2` or `Test3`. Attempting to use the _Add all missing imports_
feature results in no change to the file. If the Developer Tools are used, an error is printed, similar to the one mentioned in #40065. In fact, this is likely the root of the previous issue.

The server fails at this assert statement:
https://github.com/microsoft/TypeScript/blob/517f32940ad44933f8bb3b5d07e8c830cbfc40ea/src/services/textChanges.ts#L909-L910

(which appears to have been introduced by this commit: https://github.com/microsoft/TypeScript/commit/f4af74aae182629c6615c39325b92915e7f9ac79)

The assert statement appears to be trying to catch edits that target the same area of text. Commenting out the assert and running this modified version of the TypeScript server fixes the issue and the imports are added correctly, showing that its use, at least in this context, is unnecessary. I am not sure though of the further ramifications of removing this assert, though.

**I believe this bug happens in this case because the two imports, `Test2` and `Test3`, are adjacent to each other alphabetically, and thus the server would be placing them both in the same spot, after `Test1`. It then gets confused and thinks it's trying to add conflicting edits and stops before it can do the deed. This behavior does not exist when trying to add imports for `Test3` and `Test4` (or for  `Test1` and `Test2`), which makes me believe that the two adjacent imports have to also be sandwiched between existing imports for this bug to happen.**

_TL;DR:_
_The server fails at adding multiple imports from the same module if, when added to the existing imports list,_
- _Two or more of the imports to be added will be alphabetically adjacent to each other_
- _One or more already-existing imports will be alphabetically before all of the imports to be added_
- _One or more already-existing imports will be alphabetically after all of the imports to be added_

### 🙂 Expected behavior
It is expected that the imports be added correctly, as they do when the assert statement is commented out.
